README.md

```diff
@@ -44,3 +44,23 @@ It is possible to use a `substra-backend` skaffold file as the source for the ne
 ```
 SUBSTRA_TESTS_SKAFFOLD_FILEPATH=$SUBSTRA_SOURCE/substra-backend/skaffold.yaml make test
 ```
+
+# Test design guidelines
+
+When adding or modifying tests, please follow these guidelines.
+
+The tests suite:
+- Must be easy to configure through settings files to define the substra network to target
+- Must be independent from the substra network
+  - Substra network must be started separately
+  - Substra network can be located locally / on the cloud
+  - Substra network can be started through docker-compose or through kubernetes
+- Requires a setup with at least 2 organisations
+- Should be able to run multiple times without restarting the network
+
+Each new test:
+- Should have a deterministic behavior (not fail randomly)
+- Must be fast to run:
+  - avoid adding latency
+  - group related long running tests
```

Comment:
Here it's not clear how to do that. Maybe add or link to instructions?

Reply:
Yes, it's not very clear :)

It's most of the time a bad pattern to group tests:  one test should test a single thing.

As some of our tests are quite long to complete, I think it could be a good idea to group some of them (if it makes sense).

I'm not quite sure about this one.